test/search_test.dart

```diff
@@ -37,6 +44,70 @@ void main() {
         var result = await index.search('', limit: 3);
         expect(result.hits, hasLength(3));
       });
+
+      test('filter parameter', () async {
+        var index = await createBooksIndex();
+        var response = await index
+            .updateSettings(IndexSettings(
+              filterableAttributes: ['tag'],
+            ))
+            .waitFor();
+        expect(response.status, 'processed');
+        var result = await index.search('prince', filter: 'tag = Tale');
+        expect(result.hits, hasLength(1));
+      });
+
+      test('filter parameter with number', () async {
+        var index = await createBooksIndex();
+        var response = await index
+            .updateSettings(IndexSettings(
+              filterableAttributes: ['tag', 'book_id'],
+            ))
+            .waitFor();
+        expect(response.status, 'processed');
+        var result =
+            await index.search('', filter: 'book_id < 100 AND tag = Tale');
+        expect(result.hits, hasLength(1));
+      });
+
+      test('filter parameter with array', () async {
+        var index = await createBooksIndex();
+        var response = await index
+            .updateSettings(IndexSettings(
+              filterableAttributes: ['tag'],
+            ))
+            .waitFor();
+        expect(response.status, 'processed');
+        var result = await index.search('prince', filter: ['tag = Tale']);
+        expect(result.hits, hasLength(1));
+      });
+
+      test('filter parameter with multiple array', () async {
+        var index = await createBooksIndex();
+        var response = await index
+            .updateSettings(IndexSettings(
+              filterableAttributes: ['tag'],
+            ))
+            .waitFor();
+        expect(response.status, 'processed');
+        var result = await index.search('prince', filter: [
+          ['tag = Tale', 'tag = Tale'],
+          'tag = Tale'
+        ]);
+        expect(result.hits, hasLength(1));
+      });
+
+      test('facetDistributions parameter', () async {
+        var index = await createBooksIndex();
+        var response = await index
+            .updateSettings(IndexSettings(
+              filterableAttributes: ['tag'],
+            ))
+            .waitFor();
+        expect(response.status, 'processed');
+        var result = await index.search('prince', facetsDistribution: ['*']);
+        expect(result.hits, hasLength(2));
+      });
```

Comment:
Should we add a test on the returned facetDistribution? To test if no `0` values are present. I'm not sure it is needed but out of curiosity

Reply:
This is a test for the search engine from my POV, plus this test is already present in many SDKs